### PR TITLE
tested ExecSignUpController.js

### PR DESCRIPTION
**Pull request by saidatta993**

### CODE REVIEW

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
+const request = require('supertest')
+describe('Fetching Request', () => {
+
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:3]: Use beforeAll to include these two statements :
```
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
```
See testing/sof-116 files for reference

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    test('Return error if ngo_id is missing', async () => {
+         // signup with new user to get token.
+        const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:6]: delete 
```
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)

```

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+        const packet = {token : newloginres.body.token,ngo_id:"GAK986"};
+        const res = await request("http://localhost:8080").post('/Request/fetch').send(packet)
+        expect(res.status).toBe(404);
+        expect(res.body.message).toBe('No user requested from this ngo');
+    });
+
+    test('Return if user is successfully fetched ', async () => {
+         // signup with new user to get token.
+         const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:33]: del 
```
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
         const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)

         // login with new user to get token.
         const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
```

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token: newloginres.body.token};
+        const res = await request("http://localhost:8080").post('/Request/fetch').send(packet)
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe('Missing input data');
+    });
+
+    test('Return if no ngo_id is found in request list', async () => {
+         // signup with new user to get token.
+        const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:19]: del
```
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
 const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
```

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+        const packet = {token : newloginres.body.token,ngo_id:"GAK987"};
+        const res = await request("http://localhost:8080").post('/Request/fetch').send(packet)
+        expect(res.status).toBe(200);
+        expect(res.body.message).toBe('Requests fetched successfully');
+    });
+}
+)
+describe('storing Request', () => {
+
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:47]: use beforeAll to include:
```
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
```

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    test('Return error if ngo_id or user_id is missing', async () => {
+          // signup with new user to get token.
+          const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:50]: remove 
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
 const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
 const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token : newloginres.body.token};
+        const res = await request("http://localhost:8080").post('/Request/add').send(packet)
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe('Missing input data');
+    });
+
+    test("Return if user's request is successfully added", async () => {
+          // signup with new user to get token.
+          const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:63]: del 
const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
 const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
 const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+          const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
+  
+          // login with new user to get token.
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token :newloginres.body.token,ngo_id:"GAK987",  user_id:"6604383836a52af8f67ecc10"};
+        const res = await request("http://localhost:8080").post('/Request/add').send(packet)
+        expect(res.status).toBe(201);
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:70]: After creating fake user for testing you should delete the user.

File: server/__tests__/ExecSignup/ExecSignUp.test.js (hunk 1 above)
Comment (karthik-NvY) [reply server/__tests__/ExecSignup/ExecSignUp.test.js:3]: Make sure to also delete any test credentials created.

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        expect(res.body.message).toBe('successfully request added');
+    });
+}
+
+)
+
+describe('deleteing Request', () => {
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:77]: Same issues as above test cases.

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+  
+          // login with new user to get token.
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token : newloginres.body.token,user_id:"6604383836a52af8f67ecc11"};
+        const res = await request("http://localhost:8080").post('/Request/delete').send(packet)
+        expect(res.status).toBe(404);
+        expect(res.body.message).toBe('No Request found with the provided ID');
+    });
+
+    test('Return if user is successfully fetched ', async () => {
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:105]: You are not fetching here . This is a deleting test case.😶

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+          const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
+          const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
+  
+          // login with new user to get token.
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token :newloginres.body.token,ngo_id:"GAK987",  user_id:"6604383836a52af8f67ecc10"};
+        const res = await request("http://localhost:8080").post('/Request/add').send(packet)
+        expect(res.status).toBe(201);
+        expect(res.body.message).toBe('successfully request added');
+    });
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:72]: There should be test case when multiple user requests are made in same ngo. i.e same user making duplicate request in the ngo. This should not be allowed.

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+         // signup with new user to get token.
+        const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
+        const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
+
+        // login with new user to get token.
+        const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+
+        const packet = {token : newloginres.body.token,ngo_id:"GAK986"};
+        const res = await request("http://localhost:8080").post('/Request/fetch').send(packet)
+        expect(res.status).toBe(404);
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:27]: this is not an error. This simply means there are no users who requested for executive role.

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+  
+          // login with new user to get token.
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token : newloginres.body.token};
+        const res = await request("http://localhost:8080").post('/Request/updateRole').send(packet)
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe('Missing input data');
+    });
+
+    test('Return if no ngo_id is found in request list', async () => {
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:134]: name of test case doesn't match the test

File: server/__tests__/ExecSignup/ExecSignUp.test.js [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+          // signup with new user to get token.
+          const newuserpacket = {name:"name", password:"pass", email:"tmp@gmail.com"}
+          const newres = await request("http://localhost:8080").post('/user/signup').send(newuserpacket)
+  
+          // login with new user to get token.
+          const newloginres = await request("http://localhost:8080").post('/user/login').send(newuserpacket)
+        const packet = {token : newloginres.body.token,user_id:"6604383836a52af8f67ecc11"};
+        const res = await request("http://localhost:8080").post('/Request/updateRole').send(packet)
+        expect(res.status).toBe(404);
+        expect(res.body.message).toBe('No user found with that id');
```
Comment (karthik-NvY) [inline server/__tests__/ExecSignup/ExecSignUp.test.js:144]: in execcontroller, update role is changing the role of user to executive. It should change user in a particular NGO to executive. user_id + ngo_id

File: server/controllers/ExecSignupController.js [hunk 15]
```diff
@@ -88,13 +103,18 @@ class waitlisthandler{
...
+			return res.status(400).json({
+				success: false,
+				message: "Missing input data",
+			})};
 		try {
 			// Fetch all Task records from the database
-			const {user_id} = req.body;
-            // newDate is an object with elements as updated fields from front-end
+			
 			const updatedRole = await Roles.findOneAndUpdate(
```
Comment (karthik-NvY) [inline server/controllers/ExecSignupController.js:115]: update should happen user_id + ngo_id

File: server/server.js [hunk 16]
```diff
@@ -61,7 +61,7 @@ async function main() {
 
   app.use("/task", taskRoutes);
 
-  //app.use('/Request',ExecRoutes);
+  app.use('/Request',ExecRoutes);
```
Comment (karthik-NvY) [inline server/server.js:64]: No need for a new base route. Add exec routes in /api

**Review by karthik-NvY — CHANGES_REQUESTED** [review]

**Review by PavithranGoud — APPROVED** [review]

**Review by karthik-NvY — APPROVED** [review]

**Pull request merged**